presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoConnection.java

```diff
@@ -627,6 +634,25 @@ String getUser()
         return user;
     }
 
+    // Extra credentials consists of a list of credential name value pairs.
+    // E.g., `jdbc:presto://example.net:8080/?credentials=abc:xyz;foo:bar` will create credentials `abc=xyz` and `foo=bar`
+    Map<String, String> parseExtraCredentials(String extraCredentialString)
+    {
+        Map<String, String> extraCredentials = new HashMap<>();
+        Splitter.on(';').splitToList(extraCredentialString).stream()
+                .forEach(credential -> {
+                    List<String> nameValue = parseSingleCredential(credential);
+                    extraCredentials.put(nameValue.get(0), nameValue.get(1));
+                });
+        return extraCredentials;
```

Comment:
```
return Splitter.on(';').splitToList(extraCredentialString).stream()
    .map(this::parseSingleCredential)
    .collect(toImmutableMap(...));
```
?

Reply:
Change `EXTRA_CREDENTIALS` to be `ConnectionProperty<Map<String, String>>`, then move this inside `ConnectionProperties` as a `Converter`.

Similar to the comment in the CLI test, we should should have the conversion method validate that there are no duplicate properties by throwing `IllegalArgumentException` (see other converters).

Reply:
It's actually throwing `SQLException` instead of `IllegalArgumentException` because the `IllegalArgumentException` is caught by `AbstractConnectionProperty::getValue`.

```java
    @Override
    public Optional<T> getValue(Properties properties)
            throws SQLException
    {
        String value = properties.getProperty(key);
        if (value == null) {
            if (isRequired(properties)) {
                throw new SQLException(format("Connection property '%s' is required", key));
            }
            return Optional.empty();
        }

        try {
            return Optional.of(converter.convert(value));
        }
        catch (RuntimeException e) {
            if (value.isEmpty()) {
                throw new SQLException(format("Connection property '%s' value is empty", key), e);
            }
            throw new SQLException(format("Connection property '%s' value is invalid: %s", key, value), e);
        }
    }
```